app/views/split_checkout/_voucher_section.cable_ready.haml

```diff
@@ -0,0 +1,22 @@
+%div#voucher-section
+  = form_with url: checkout_update_path(:payment), model: order, method: :put, data: { remote: "true" } do |f|
+    - if order.distributor.vouchers.present?
+      .checkout-title
+        = t("split_checkout.step2.voucher.apply_voucher")
+
+      .checkout-input
+        .two-columns-inputs.voucher
+          - if voucher_adjustment.present?
+            %span.button.voucher-added
+              %i.ofn-i_051-check-big
+                = t("split_checkout.step2.voucher.voucher", voucher_amount: voucher_adjustment.originator.display_value)
+            = link_to t("split_checkout.step2.voucher.remove_code"), voucher_adjustment_path(id: voucher_adjustment.id), method: "delete", data: { confirm: t("split_checkout.step2.voucher.confirm_delete") }
+            - # TODO: this might not be true, ie payment method include fee which wouldn't be covered by voucher, tax implication raise total to be bigger than voucher other ?
+            - if voucher_adjustment.originator.value > order.total
+              .checkout-input
+                %span.formError.standalone
+                  = t("split_checkout.step2.voucher.warning_forfeit_remaining_amount")
+          - else
+            = f.text_field :voucher_code, { placeholder: t("split_checkout.step2.voucher.placeholder"), class: "voucher" }
+            - # TODO: enable button when code entered
```

Comment:
Not in the requirement, but I was thinking of disabling the "Apply" button when nothing is entered in the input, I never got around to do it.
I just played around with it, and I should probably do that, as right now if you click "Apply" with a blank code, it will try to go the order summary steps, unexpected behaviour !

 I am wondering if there is already something like this implemented in the code base ?

Reply:
What about input validation `required: true`? Then you can't submit until the value is entered.

Reply:
That would have been too easy ! The `required: true` prevent the user from going to the next step if they don't have a voucher to use.
I ended up adding a small Stimulus controller to enable/disable the "Apply" button.

Reply:
Oh, I imagined two independent forms... but good that you found a solution.